utils/frame/benchmarking-cli/src/machine/hardware.rs

```diff
@@ -18,8 +18,55 @@
 //! Contains types to define hardware requirements.
 
 use lazy_static::lazy_static;
-use serde::{Deserialize, Serialize};
-use std::fmt;
+use sc_sysinfo::Throughput;
+use serde::{
+	de::{MapAccess, Visitor},
+	ser::SerializeMap,
+	Deserialize, Deserializer, Serialize, Serializer,
+};
+use sp_std::{fmt, fmt::Formatter};
+
+pub fn serialize_throughput<S>(t: &Throughput, serializer: S) -> Result<S::Ok, S::Error>
+where
+	S: Serializer,
+{
+	let mut map = serializer.serialize_map(Some(1))?;
+	let (value, unit) = t.normalize();
```

Comment:
Hmm.... I'm a little spooked by using this here. It's not obvious in the `sysinfo` crate that the exact string returned from `normalize` is used for serialization/deserialization here (especially since over there it's only used for formatting), so I can imagine someone could e.g. change how the units are called or the formatting slightly over there and it'd break serialization here.

Could we maybe change `normalize` to return an *enum* instead? e.g. get it to return something like this instead of a string:

```
enum Unit {
    GiBs,
    MiBs,
    KiBs
}
```